### PR TITLE
Refactor RoomHeader, and fix topic updates

### PR DESCRIPTION
**Pull request by richvdh**
This is in aid of fixing https://github.com/vector-im/vector-web/issues/1299,
which is actually trivial once you can see what's going on.

Suggest looking at the individual commits to review:
* https://github.com/matrix-org/matrix-react-sdk/commit/70133ae661d3e9603c8390bbdd5d36a1db3a6d13?w=1
* https://github.com/matrix-org/matrix-react-sdk/pull/252/commits/6422f7ecc8edc3e053cb20a2b3117c515d408914
* https://github.com/matrix-org/matrix-react-sdk/pull/252/commits/0d261c37c2ede83eb7a45f4721694d4ee299510f

### CODE REVIEW

File: src/components/structures/CreateRoom.js [hunk 1]
```diff
@@ -249,13 +249,15 @@ module.exports = React.createClass({
...
-            var RoomHeader = sdk.getComponent("rooms.RoomHeader");
+            var SimpleRoomHeader = sdk.getComponent("rooms.SimpleRoomHeader");
 
             var domain = MatrixClientPeg.get().getDomain();
 
             return (
             <div className="mx_CreateRoom">
-                <RoomHeader simpleHeader="Create room" />
+                <SimpleRoomHeader>
+                    Create room
```
Comment (ara4n) [inline src/components/structures/CreateRoom.js]: Hm, I'm quite unconvinced by passing in the title as the contents of the tag.  I'd expect simple room headers to have structure for (optional) avatar and title at the least - at which point i'd expect to be passing them in as properties.  Is there a rationale i'm missing?

**Comment by ara4n** [comment]
lgtm other than minor query over SimpleRoomHeader's API - assuming that you've checked that the editing still works; the EditableText impl (via contentEditable) has been a bit fragile, but should be okay now, in terms of correctly keeping the editable text contents in sync with the state of the parent components.  But if you haven't reintroduced the bug where the EditableTexts forget their state whenever you re-render() their parent, lgtm

File: src/components/structures/CreateRoom.js (hunk 1 above)
Comment (richvdh) [inline src/components/structures/CreateRoom.js]: Only that it looked like the right thing to do given the structure of SimpleRoomHeader. I agree it doesn't lead to a nice api. I'll change it.

**Comment by richvdh** [comment]
It seems to work...

**Pull request merged**